### ISSUE TITLE
Mark PRs merged by bors as merged not closed

### ISSUE THREAD

**Issue by eric-wieser** [bors-ng/bors-ng#1217]
Right now (at least with `use_squash_merge = true`), PRs merged by bors end up with `[Merged by Bors]` in the title, but are still considered "closed" instead of "merged" by github. This has a number of downsides:

* Repository statistics about merged vs closed aren't useful
* The "revert PR" button doesn't appear
* I don't think "Closes #NNN" works in PR descriptions, since github considers the PR abandoned not resolved

I suspect there are technical limitations on the github side ([this thread?](https://github.community/t/feature-request-mark-pull-request-as-merged/14194) that make this impossible, but it would be great if this limitation could be documented. The bors website shows everything working normally, so I guess this only comes up in certain configurations.

**Comment by cart** [comment]
Yup this is causing issues for us in Bevy. We want clean squashed history without bors merge commits, so we enable `use_squash_merge`, but this comes at the cost of the problems mentioned above (broken stats are a huge loss for us). We also want to distinguish between merged state and closed state in our project boards, but right now that is impossible.

**Comment by DJMcNab** [comment]
I think this is related to https://github.com/bors-ng/bors-ng/issues/899.

A few of us (@bjorn3, @mockersf ) in the [bevy discord](discord.gg/bevy) have been theorising a potential way to workaround this. 

In this scheme bors would act as it currently does, until it reaches the point where the PR would currently be closed. At that point, it would change the base branch of the PR to its own version of the target branch (say a branch with name `bors/main`). It would update that branch to the head commit of main (either before or after the relevant PR is merged, as https://github.com/bors-ng/bors-ng/issues/899#issuecomment-596075511 suggests either would work). It would then use the GitHub API to merge the PR into that branch, which marks the PR as merged.

**Comment by yesthesoup** [comment]
Have run into this issue as well for our configuration. Just enabled `use_squash_merge` in the repo's `bors.toml`, but it's now closing all the merged PRs instead of merging them as before.
